source/adapters/cuda/event.cpp

```diff
@@ -42,11 +42,32 @@ ur_event_handle_t_::ur_event_handle_t_(ur_context_handle_t Context,
   urContextRetain(Context);
 }
 
+void ur_event_handle_t_::reset() {
+  RefCount = 0;
```

Comment:
Is this thread safe?

Reply:
This became obsolete.
RefCount is already 0 at this point.
Making this assignment redundant.
I replaced it with an assertion to ensure that we're not resetting an event that still holds references.